hazelcast/src/main/java/com/hazelcast/util/executor/HazelcastManagedThread.java

```diff
@@ -73,15 +88,26 @@ protected void afterRun() {
 
     /**
      * Manages the thread lifecycle and can be overridden to customize if needed.
+     * Manages the thread lifecycle.
      */
-    public void run() {
+    public final void run() {
```

Comment:
Without all the superfluous abstraction this can be simplified to something like this
```
 public final void run() {
     String pool = getPoolName();
        if(ThreadAffinity.ENABLED && pool!=null){
            String cpu = ThreadAffinity.getCPU(pool);
            ThreadAffinity.setAffinity(cpu);
            try{
                run();
            }finally{
                 ThreadAffinity.returnCPuPool(pool);
            }
        }else{
            run();
        }
```

Reply:
And you make threadAffinity the HZ wrapper around the thread affinity library from Peter Lawrey.